### ISSUE TITLE
[rollout issue] Arcade publishing broken as of latest Arcade-services deployment 

### ISSUE THREAD

**Issue by riarenas** [dotnet/arcade#9471]
### Build

https://dnceng.visualstudio.com/internal/_build/results?buildId=1789965&view=logs&j=226748d0-f812-5437-d3f0-2dd291f5666e&t=bad11196-972e-5d03-45a8-9db526506073

https://dnceng.visualstudio.com/internal/_build/results?buildId=1789677&view=logs&j=29d4eac0-e772-56a4-ad0d-2d727d908ceb&t=4edcb01d-2da8-54c4-439e-584d3c1a5e90

https://dnceng.visualstudio.com/internal/_build/results?buildId=1789315&view=logs&s=0d7db6e4-c533-50c8-77c7-b1b9fc049ed6&j=226748d0-f812-5437-d3f0-2dd291f5666e

### Action required for the engineering services team

To triage this issue (First Responder / @dotnet/dnceng):
- [ ] Open the failing build above and investigate
- [ ] Add a comment explaining your findings 

If this is an issue that is causing build breaks across multiple builds and would get benefit from being listed on the [build analysis check](https://github.com/dotnet/arcade/blob/174acbcafeba3ae339c1935ec48812514ee09837/Documentation/Projects/Build%20Analysis/Introduction.md), follow the next steps:
1. Add the label "Known Build Error"
2. Edit this issue and add an error string in the Json below that can help us match this issue with future build breaks. You should use the [known issues documentation](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-message-section)

 ```json
 {
    "ErrorMessage" : "The framework 'Microsoft.NETCore.App', version '6.0.0' was not found."
 }
 ```

### Additional information about the issue reported

I believe this was caused by the upgrade of all of arcade-services to use .net 6. The latest version of darc requires a 6.0 runtime to be installed in the machine.

**Comment by adiaaida** [comment]
We saw this in dotnet-release as well. We need to make sure that arcade's build for publishing downloads dotnet because the agents don't already have it installed

**Comment by adiaaida** [comment]
Also make the agents have dotnet 6.0 installed

**Comment by riarenas** [comment]
We can't keep all builds on the floor while that change propagates through arcade though.

**Comment by adiaaida** [comment]
@MattGal is dotnet installed on the buildpool machines via VS/VS workloads by default? The machine obviously has 2.1, 3.1 and 5.0 installed globally, but not 6.0.

**Comment by riarenas** [comment]
@alexperovich had the idea of hardcoding the latest working version of darc to the darc-version API and hotfix the service with that hack while we get the right runtimes installed by this job. Does anyone have any better ideas @mmitche @ChadNedzlek @dotnet/dnceng

**Comment by MattGal** [comment]
> @MattGal is dotnet installed on the buildpool machines via VS/VS workloads by default? The machine obviously has 2.1, 3.1 and 5.0 installed globally, but not 6.0.

[windows-2022 / 1es-windows-2022](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2022-Readme.md) has 6.0.300 SDK installed.   Nothing else I'm aware of comes with it "in box"

**Comment by adiaaida** [comment]
Sorry, I meant do the other images have 2.1, 3.1 and 5.0 installed by default?

**Comment by riarenas** [comment]
One potential way to fix this is to add a "use dotnet" task to install the 6.0 runtime as part of the publishing stage, but we would have to make sure all servicing branches get the updated template, which is probably the biggest hurdle with that solution. We would need to make sure every repo in every relevant branch has received the update before changing the darc-version api back to what it used to be.

**Comment by riarenas** [comment]
The hotfix with the workaround by basically pinning the version of darc in every repo seems to be working. We still need to figure out the steps for a proper fix.

**Comment by mmitche** [comment]
Could we publish darc as self-contained?

**Comment by riarenas** [comment]
Ooh, yeah that would bring the necessary runtime along with it, but I believe we would need to produce both a linux and windows version of the tool? (I believe we've already said that darc doesn't really work on macos)

**Comment by premun** [comment]
Can we just multi-target to .NET 5.0 as well if that one is available? That might be the easiest if we need to unblock something

**Comment by adiaaida** [comment]
With the changes to arcade services to specify win-x64, honestly, I don't think darc currently works on anything other than Windows (or at least, we don't guarantee that it does).

**Comment by adiaaida** [comment]
I would also suggest figuring out how the other dotnet versions are already installed on the images we used and look into if a) they even should be and b) if they should, why 6.0 isn't also installed... and then make sure that it is.

**Comment by mmitche** [comment]
5.0 goes out of public support this month, so I'd also expect it to disappear from the 1ES images soon.

**Comment by adiaaida** [comment]
I talked to Matt. We basically copy what's in the windows-2019 queue for what version of VS and therefore what dotnet versions are installed on the machines. 2019 only has 5.0 and below installed by default, so... that's what we've got too on the 2019 buildpool queue. Alas, it doesn't sound like an easy "oh just update the images" solution.

**Comment by riarenas** [comment]
I don't think we should rely on what's on the image in this case. It's the runtime, we should be able to install whichever ones we need (or make it self contained so that we don't need an external runtime)

**Comment by adiaaida** [comment]
I think my preference is to go self-contained. Anytime we have to update arcade-services to a new dotnet version, we'd get caught in this trap. Better to make darc come with everything it needs

**Comment by mmitche** [comment]
I'm guessing they install .NET Core 6 on the 2022 images.

**Comment by adiaaida** [comment]
They do

**Comment by ChadNedzlek** [comment]
It would probably be "easier" to just publish the tool X times, rather than try to zombie up a package. "darc-win", "darc-macos", that sort of thing. Since at that point it's just a switch to the publish command.

**Comment by dougbu** [comment]
Why is this issue still open❔ I thought this problem was behind us

**Comment by ChadNedzlek** [comment]
The change we made is a huge hack (basically just block any future updates to the darc tool).  We need a real solution that isn't "never make any changes ever again", unfortunately.

And none of those options are looking good right now.  Though perhaps that should be handled with another issue that isn't FR/hotfix/RCA'd.

I'm gong to close this issue in favor of that.
